### PR TITLE
update(panel): choose first positioning that is on-screen.

### PR DESCRIPTION
**Pull request by gmoothart**
Looks through all positions supplied by the user, and picks the first one which is on-screen. If none are on-screen, the last one will be chosen. Part of #7878

@ErinCoughlan @KarenParker @ThomasBurleson @jelbourn @DerekLouie

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -1722,14 +1731,36 @@ MdPanelPosition.prototype.getTransform = function() {
...
+ * @return {boolean}
+ */
+MdPanelPosition.prototype._isOnscreen = function(panelEl) {
+  // this works because we always use fixed positioning for the panel,
+  // which is relative to the viewport.
+  // TODO(gmoothart): take into account _translateX and _translateY to the
+  //   extent feasible.
+
+  var left = parseInt(this.getLeft(panelEl));
+  var top = parseInt(this.getTop(panelEl));
```
Comment (KarenParker) [inline src/components/panel/panel.js]: These don't take any parameters.

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -1763,6 +1795,29 @@ MdPanelPosition.prototype._calculatePanelPosition = function(panelEl) {
   // TODO(ErinCoughlan): Update position on scroll.
   // TODO(ErinCoughlan): Position panel intelligently to keep it on screen.
 
+  if (this._actualPosition) {
+    this._calculatePanelPosition(panelEl, this._actualPosition);
+    return;
+  }
+
+  for(var i=0; i < this._positions.length; i++) {
```
Comment (ErinCoughlan) [inline src/components/panel/panel.js]: nit: space after `for` and between the equals

**Comment by ErinCoughlan** [comment]
LGTM (I'm not looking forward to these merge conflicts)

**Comment by KarenParker** [comment]
👍

**Pull request merged**